sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -615,6 +633,30 @@ def _construct_from_components(self, components):
     return tuple(components)
 
 
+class _ConcatSequence(object):
+  def __init__(self, head, tail):
+    self._head = head
+    self._tail = tail
+
+  def __iter__(self):
+    for elem in self._head:
+      yield elem
+    for elem in self._tail:
+      yield elem
+
+  def __eq__(self, other):
+    return list(self) == list(other)
+
+  def __hash__(self):
+    raise NotImplementedError
+
+  def __reduce__(self):
+    return list, (list(self),)
+
+
+FastPrimitivesCoderImpl.register_iterable_like_type(_ConcatSequence)
```

Comment:
How come elsewhere we do

coder_impl.FastPrimitivesCoderImpl...

but here just

FastPrimitivesCoderImpl...

Reply:
Because FastPrimitivesCoderImpl is local to this module.